guillotina/db/storages/pg.py

```diff
@@ -782,21 +782,21 @@ def objects_table_name(self):
 
     @restart_conn_on_exception
     async def open(self):
-        conn = await self.pool.acquire(timeout=self._conn_acquire_timeout)
+        conn = await self.pool.acquire()
         return conn
 
     async def close(self, con):
         try:
-            await shield(asyncio.wait_for(self.pool.release(con, timeout=1), 1))
+            await shield(self.pool.release(con, timeout=1))
         except (
             asyncio.CancelledError,
-            RuntimeError,
             asyncio.TimeoutError,
             asyncpg.exceptions.ConnectionDoesNotExistError,
         ):
-            pass
+            log.warning("Exception on connection close", exc_info=True)
```

Comment:
Should we also add fallback handler to terminate?

I suppose we need to be ultra careful to always release connection.

Reply:
This is the fallback handler: https://github.com/plone/guillotina/pull/690/files#diff-cf6270d3de166f3eb7ec0d3cce4e4ac2R140